packages/remix-server-runtime/server.ts

```diff
@@ -286,19 +286,7 @@ async function handleDataRequestRR(
     return response;
   } catch (error) {
     if (error instanceof Response) {
-      // To match existing behavior of remix-thrown responses,
-      // we construct a new one here with a null body and just
-      // the required headers. No remix-throw that surface to
-      // this point will ever have a body. This contrasts with
-      // the new router implementation that has a body under
-      // some conditions.
-      return new Response(null, {
-        status: error.status,
-        statusText: error.statusText,
-        headers: {
-          "X-Remix-Catch": "yes",
-        },
-      });
```

Comment:
@jacob-ebey This might break integration tests - unsure what prompted it in the first place but the new Response creation here was causing mismatches in response `content-type` headers in the `server.ts` unit tests :/

If having a body is problematic we can also look into removing the body from the router Response.  I _think_ the only one with a body now might be the the 405 no action found response but we may not be calling that anymore now that `callRouteAction` is conditional?